src/nvim/api/vim.c

```diff
@@ -888,7 +888,7 @@ void nvim_set_current_buf(Buffer buffer, Error *err)
 {
   buf_T *buf = find_buffer_by_handle(buffer, err);
 
-  if (!buf) {
+  if (!buf || curwin->w_buffer == buf) {
     return;
```

Comment:
what happens if we just delete the `w_p_wfb ` check below? `do_buffer` already checks it. then this problem wouldn't have happened. @ColinKennedy

Reply:
The unittests are fairly thorough and this PR covers the other cases. As long as the tests all pass we should be good.

Reply:
I personally prefer the "add the check at the caller instead of in the core" because if gives callers more control but we shouldn't double check if it can be avoided.

Reply:
https://github.com/neovim/neovim/pull/31595